### PR TITLE
implemented repositories, services, added Book model

### PR DESCRIPTION
**Pull request by SaiyaJ1N**

### CODE REVIEW

File: src/main/java/com/onlinebookshop/exception/DataProcessingException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.onlinebookshop.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String msg, Throwable ex) {
```
Comment (DmytroV95) [inline src/main/java/com/onlinebookshop/exception/DataProcessingException.java]: Not informative argument names

File: src/main/java/com/onlinebookshop/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
+public class Book {
```
Comment (DmytroV95) [inline src/main/java/com/onlinebookshop/model/Book.java:15]: You can specify a table name annotation.

File: src/main/java/com/onlinebookshop/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
```
Comment (DmytroV95) [inline src/main/java/com/onlinebookshop/repository/impl/BookRepositoryImpl.java:18]: Annotation is not necessary here.  Spring will understand it anyway.

**Review by DmytroV95 — CHANGES_REQUESTED** [review]
GJ! Let's improve your solution.

File: src/main/java/com/onlinebookshop/repository/impl/BookRepositoryImpl.java (hunk 3 above)
Comment (SaiyaJ1N) [reply src/main/java/com/onlinebookshop/repository/impl/BookRepositoryImpl.java:18]: Yes, thank you for review, i know that Spring will automatically autowire construcrute if we have only 1 in our Entity. But in my opinion its more clearly, explicitly when were using annotation.

**Review by DmytroV95 — APPROVED** [review]
LGTM!

File: pom.xml [hunk 4]
```diff
@@ -23,11 +23,38 @@
             <artifactId>spring-boot-starter</artifactId>
         </dependency>
```
Comment (vovasalyha) [inline pom.xml]:
```suggestion
```

File: pom.xml [hunk 5]
```diff
@@ -23,11 +23,38 @@
...
 
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
```
Comment (vovasalyha) [inline pom.xml]: It's better to put test dependencies at the end.

File: pom.xml [hunk 6]
```diff
@@ -23,11 +23,38 @@
...
+            <artifactId>lombok</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.33</version>
+        </dependency>
+
+        <dependency>
```
Comment (vovasalyha) [inline pom.xml:48]: Why did you add dependency which you don't need now?

File: src/main/java/com/onlinebookshop/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import com.onlinebookshop.repository.BookRepository;
+import com.onlinebookshop.service.BookService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (vovasalyha) [inline src/main/java/com/onlinebookshop/service/impl/BookServiceImpl.java]: In most cases it's considered better approach to use constructor injection (there are some exceptions though)

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ! Let's improve your solution a bit.

**Review by vovasalyha — APPROVED** [review]
GJ!
